### PR TITLE
feat(issue-3640):user should be able to change config in cypress GUI

### LINKED ISSUE #3640: Feature Request: User should be able to change config in cypress GUI interface
### Current behavior:

Configuration can only be passed either by code or cli 

### Desired behavior:

Configuration can be modified through GUI interface

A use case would be:

We have a test suite that checks if all the strings on our web app are translated. 

The test code is the same for any given languages, all we change is the config parameter to select what languages we want to run with. 

Currently, I have to change the config file and restart GUI. It would be great if I can just change config on the GUI and run the test. 

### Versions

3.1.5

### PR DESCRIPTION
**Pull request by Manuel-Suarez-Abascal**
Closes #3640
### User facing changelog
Added functionality for the user to change config values in the Cypress GUI interface.

### Additional details
Currently, we can change `cypress.json` config values through the code or CLI params. We want to able to change the values through the GUI as well. I have added an edit icon which you can double click & an input will show up to change the current config value in the Cypress GUI.

### How has the user experience changed?

https://user-images.githubusercontent.com/32891808/113575052-4ad10e80-95eb-11eb-8b80-048c0ce21037.mov

### PR Tasks
- [ ] Have tests been added/updated? (to be determined?)

**Other details:**

@jennifer-shehane perhaps you can offer me some assistance with this feature. I might need some onboarding on how to change the app's state in this codebase. I can see you use `mobx` as state management. Can you point on how you handle/mutated the state on this codebase?

- This is a **work in progress** PR. I was able to figure out how to change the config values in the labels, but it doesn't change in the app's state( from the observer?). Please refer to the gif above for better context.

- Any comments on the styling, implementation to details?

- Should I add tests to this feature? If so, what should they cover & in which file should I include them?

Thanks! 🙏

### CODE REVIEW

File: packages/desktop-gui/src/settings/configuration.jsx [hunk 1]
```diff
@@ -48,7 +48,28 @@ const normalizeWithoutMeta = (value = {}) => {
...
+  }
+  const [isEditable, setIsEditable] = useState(false)
+
+  const [value, setValue] = useState('')
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    setIsEditable(!isEditable)
+
+    formattedData = value
```
Comment (Manuel-Suarez-Abascal) [inline packages/desktop-gui/src/settings/configuration.jsx:69]: @jennifer-shehane I have tried directly changing the `formattedData`'s state here. However, it won't work. It's probably related to how to mutate state using `mobx` which I'm not familiar with plus there is probably an abstraction on top of it in this codebase. 

Can you point to where/how to change the state of the config values?

File: packages/desktop-gui/src/settings/configuration.jsx [hunk 2]
```diff
@@ -48,7 +48,27 @@ const normalizeWithoutMeta = (value = {}) => {
...
+  const [isEditable, setIsEditable] = useState(false)
+
+  const [value, setValue] = useState('')
+
+  const [configData, setConfigData] = useState(formattedData)
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    setIsEditable(!isEditable)
+    setConfigData(value)
```
Comment (Manuel-Suarez-Abascal) [inline packages/desktop-gui/src/settings/configuration.jsx:70]: I'm able to change config values here, but it only takes effect in the UI labels and not in the app's config state. I'm assuming I need to change it before it gets passed here `<ConfigDisplay data={project.resolvedConfig} />`? 

Can you(or somebody on the team) onboard/ guide me on how to use `hooks` to change the `observer`?

File: packages/desktop-gui/src/settings/configuration.jsx [hunk 3]
```diff
@@ -50,7 +50,29 @@ const normalizeWithoutMeta = (value = {}) => {
 const ObjectLabel = ({ name, data, expanded, from, isNonenumerable }) => {
   const formattedData = formatValue(data)
 
-  return (
+  const localData = useLocalStore(() => ([formattedData]))
```
Comment (Manuel-Suarez-Abascal) [inline packages/desktop-gui/src/settings/configuration.jsx:53]: It keeps changing the values in the UI, but the actual app's state is still the same. I tried importing these methods from `mobx-react`, but I haven't figured it out yet. Any tips will be appreciated. [Source](https://mobx-react.js.org/state-local#what-about-global-store)

**Comment by flotwig** [comment]
@Manuel-Suarez-Abascal thank for for opening a PR. This looks good so far, although, as you note, it does not actually propagate the config changes all the way to `cypress.json` and the rest of the app state.

However, let's pause on this effort for now - at some point in the future, we may end up replacing `cypress.json` with a more flexible `cypress.js`, which will require a rethinking of this feature. As a result, this PR isn't something we can merge in the near future.

**Comment by Manuel-Suarez-Abascal** [comment]
> @Manuel-Suarez-Abascal thank you for opening a PR. This looks good so far, although, as you note, it does not actually propagate the config changes all the way to `cypress.json` and the rest of the app state.
> 
> However, let's pause on this effort for now - at some point in the future, we may end up replacing `cypress.json` with a more flexible `cypress.js`, which will require a rethinking of this feature. As a result, this PR isn't something we can merge in the near future.

Thanks for your reply @flotwig. I'll go ahead then & close the PR for now. Whenever we're ready to work on this feature after the refactoring to `cypress.js` file. We can go back to it & finish it!

**Pull request closed without merging**